lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -550,6 +554,51 @@ def _graded_scorable_blocks_to_header(cls, course):
 
 
 class ProblemResponses(object):
+
+    @classmethod
+    def _build_problem_list(cls, course_blocks, root, path=None):
+        """
+        Generate a tuple of display names, block location paths and block keys
+        for all problem blocks under the ``root`` block.
+        """
+        display_name = course_blocks.get_xblock_field(root, 'display_name')
+        if path is None:
+            path = [display_name]
+        yield display_name, path, root
+        for block in course_blocks.get_children(root):
+            display_name = course_blocks.get_xblock_field(block, 'display_name')
+            if block.block_type == 'problem':
+                yield display_name, path, block
+            else:
+                for result in cls._build_problem_list(course_blocks, block, path + [display_name]):
+                    yield result
+
+    @classmethod
+    def _build_student_data(cls, user_id, course_id, problem_location):
+        """
+        Generate a list of problem responses for all problem under the
+        ``problem_location`` root.
+        """
+        problem_key = UsageKey.from_string(problem_location).map_into_course(course_id)
+        user = get_user_model().objects.get(pk=user_id)
+        course_blocks = get_course_blocks(user, problem_key)
+
+        student_data = []
+        max_count = settings.FEATURES.get('MAX_PROBLEM_RESPONSES_COUNT')
+        for title, path, block in cls._build_problem_list(course_blocks, problem_key):
+            responses = list_problem_responses(course_id, block, max_count)
+            student_data += responses
+            for response in responses:
+                response['title'] = title
+                response['location'] = ' > '.join(path)
```

Comment:
I noticed that this returns inconsistent results when using root nodes of different depths. For example, if you put in a sequential ID and generate a report, this will say e.g. "Homework - Question  Styles > Multiple Choice Questions" but if putting the block ID directly, the 'location' is just "Multiple Choice Questions". That's probably fine though, as I think always making the location relative to the root could be too inefficient. And the point is to resolve ambiguity within the file, which is done.